src/main/java/org/javarosa/core/model/condition/LifecycleSignaler.java

```diff
@@ -0,0 +1,42 @@
+package org.javarosa.core.model.condition;
+
+/**
+ * An object which can be passed through a call chain to portably keep track of the abandoned
+ * status of a lifecycle.
+ *
+ * Created by ctsims on 9/28/2017.
+ */
+
+public class LifecycleSignaler implements Abandonable {
+    private boolean isAbandoned = false;
+
+    /**
+     * If the current lifecycle has been abandoned, this will throw a
+     * RequestAbandonedException in order to terminate the current process
+     */
+    @Override
+    public void assertNotAbandoned() {
+        if(isAbandoned) {
+            throw new RequestAbandonedException();
+        }
+    }
+
+    @Override
+    public void signalAbandoned() {
+        this.isAbandoned = true;
+    }
+
+    public static void AssertNotAbandoned(Abandonable abandonable) {
```

Comment:
why is the first letter of the method name capitalized? here and below

Reply:
These are static methods, so the name is capitalized to differentiate it from the class method.